stdlibs/stdlibs.go

```diff
@@ -275,6 +275,48 @@ func InjectPackage(store gno.Store, pn *gno.PackageNode) {
 				m.PushValue(res0)
 			},
 		)
+		pn.DefineNative("ExecAsPkg",
+			// TODO: rename UnsafeExecAsPkg?
+			gno.Flds( // params
+				"fn", gno.FuncT(nil, nil),
+			),
+			gno.Flds( // results
+			),
+			func(m *gno.Machine) {
+				// TODO: limit stack size: 1? 32?.
+				// TODO: limit usage in goroutines?
+				// TODO: build a new sub-context instead of patching the root one.
+				//       to support multiple concurrent contexts and advanced flows.
+				arg0 := m.LastBlock().GetParams1().TV
+				fn := arg0.GetFunc()
+
+				ctx := m.Context.(ExecContext)
+				ctx.OrigCaller = ctx.OrigPkgAddr
+
+				backupContext := m.Context.(ExecContext)
+
+				m.PushValue(gno.TypedValue{
+					T: &gno.InterfaceType{},
+					V: gno.ContextValue{
+						Context: backupContext,
+					},
+				})
+				m.PushOp(gno.OpSetContext)
+
+				cx := gno.Call(gno.FuncT(nil, nil))
+				m.PushValue(gno.TypedValue{})
```

Comment:
We are still unsure why we need to add this useless value in the values' stack. The thing is if we don't, the stack is truncated by one value (when `PopFrameReset` is invoked). In that case, the truncated value is the context value we pushed earlier, and `doOpSetContext` cannot work properly without it.

We'll try to figure out why this value is truncated.